### PR TITLE
Add support for generating Dockerfile.c9s

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for generation Dockefile.c9s files.
And specify distgen configuration file.

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
The configuration file for C9S already exist here: https://github.com/devexp-db/distgen/blob/main/distgen/distconf/centos-stream-9-x86_64.yaml

**Comment by phracek** [comment]
[test]

**Comment by pkubatrh** [comment]
LGTM

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: generate.sh [hunk 1]
```diff
@@ -84,6 +84,10 @@ parse_rules() {
                     if ! [[ "$DG_CONF" =~ fedora-[0-9]{,2}-x86_64.yaml ]]; then
                         continue
                     fi
+                elif [[ "$dest" == *"Dockerfile.c9s" ]]; then
```
[the remark this answers is not in the document]
Comment (phracek) [reply generate.sh:87]: https://acloudguru.com/blog/engineering/conditions-in-bash-scripting-if-statements
 especially double brackets syntax

[the remark this answers is not in the document]
Comment (zmiklank) [reply generate.sh:87]: and do we want different behavior for different Dockerfiles?

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: generate.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (phracek) [reply generate.sh:87]: From my point of view, I would prefer to have strict files and not *"Dockerfile.fedora".

But I did not test what will change.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Fedora test failed here:
 ```bash

08:36:11                 out:  [PASSED] for 'hw' test_run_hw_application
08:36:11                 out:  [PASSED] for 'hw' test_incremental_build
08:36:11                 out:  [PASSED] for 'hw' test_build_express_webapp
08:36:11                 out:  [PASSED] for 'clients' express
08:36:11                 out:  [FAILED] for 'clients' pino
08:36:11                 out:  [PASSED] for 'clients' prom-client
08:36:11                 out:  [PASSED] for 'binary' test_run_binary_application
```

**Comment by phracek** [comment]
CentOS 7 test failed here:
```bash
1m[INFO] Testing variable presence during `docker exec`[0m

 Value /opt/rh/rh-nodejs14/root/usr/share/man is missing from variable MANPATH
MANPATH=/opt/rh/rh-python27/root/usr/share/man:/opt/rh/python27/root/usr/share/man:/opt/rh/httpd24/root/usr/share/man:/opt/rh/rh-nodejs14/root/usr/share/man

[1m[INFO] Checking if all scl variables are defined in Dockerfile[0m

```

**Comment by phracek** [comment]
RHEL7 test failed here:
```bash
08:50:02                 out:  [PASSED] for 'app' test_scl_usage
08:50:02                 out:  [PASSED] for 'app' test_connection
08:50:02                 out:  [PASSED] for 'app' test_docker_run_usage
08:50:02                 out:  [FAILED] for 'app' test_scl_variables_in_dockerfile
08:50:02                 out:  [PASSED] for 'app' test_npm_functionality
08:50:02                 out:  [PASSED] for 'app' test_npm_cache_cleared
```

```bash
08:39:51                 out: 
08:39:51                 out: [INFO] Running test test_scl_variables_in_dockerfile .......
08:39:51                 out: 
08:39:51                 out: Testing npm availability in Dockerfile
08:39:51                 out: 
                    Sending build context to Docker daemon 2.048 kB
                    
08:39:51                 out: Step 1/2 : FROM rhscl/nodejs-14-rhel7:1
08:39:51                 out:  ---> 5ac5d773b0df
08:39:51                 out: Step 2/2 : RUN command -v npm | grep "^/opt/rh"
08:39:51                 out:  ---> Running in 984da1c20b29
08:39:51                 out: 
08:39:51                 out: /opt/rh/rh-nodejs14/root/usr/bin/npm
08:39:51                 out:  ---> a226d6d877e0
08:39:51                 out: Removing intermediate container 984da1c20b29
08:39:51                 out: Successfully built a226d6d877e0
08:39:51                 out: 
08:39:51                 out: [INFO] Testing variable presence during `docker exec`...
08:39:51                 out: 
08:39:52                 out:  Value /opt/rh/rh-nodejs14/root/usr/share/man is missing from variable MANPATH
08:39:52                 out: MANPATH=/opt/rh/rh-nodejs14/root/usr/share/man
08:39:52                 out: S2I image 'rhscl/nodejs-14-rhel7:1' test FAILED (exit code: 1)
```

**Pull request merged**